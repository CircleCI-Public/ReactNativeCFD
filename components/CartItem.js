import * as React from 'react';
import {IconButton, Paragraph} from 'react-native-paper';
import {StyleSheet, View} from 'react-native';
import menuData from '../data/MenuData';

const styles = StyleSheet.create({
  Container: {
    display: 'flex',
    flexDirection: 'row',
  },
  ItemContainer: {
    flexDirection: 'column',
  },
});

const CartItem = ({menuItem, removeFromCart, index}) => {
  return (
    <View style={styles.Container}>
      <IconButton
        icon="delete"
        onPress={() => {
          removeFromCart(index);
        }}
      />
      <View style={styles.ItemContainer}>
        <Paragraph style={{marginBottom: 0}}>{menuItem.title}</Paragraph>
        <Paragraph style={{fontSize: 12, marginTop: 0}}>
          {menuItem.description}
        </Paragraph>
        <Paragraph style={{fontSize: 14}}>${menuItem.cost}</Paragraph>
      </View>
    </View>
  );
};

export default CartItem;
