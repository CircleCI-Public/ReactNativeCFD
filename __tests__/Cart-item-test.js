import React from 'react';
import renderer from 'react-test-renderer';
import CartItem from '../components/CartItem';

test('renders correctly', () => {
  const tree = renderer.create(<CartItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
