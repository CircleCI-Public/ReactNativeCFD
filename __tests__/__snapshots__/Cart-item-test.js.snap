// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "margin": 6,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 18,
            "height": 36,
            "width": 36,
          },
          undefined,
          undefined,
        ],
      ]
    }
  >
    <View>
      <Text
        accessibilityElementsHidden={true}
        allowFontScaling={false}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontSize": 24,
            },
            Array [
              Object {
                "lineHeight": 24,
                "transform": Array [
                  Object {
                    "scaleX": 1,
                  },
                ],
              },
              Object {
                "backgroundColor": "transparent",
              },
            ],
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ó°†´
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              Object {
                "marginBottom": 0,
              },
            ],
          ],
        ]
      }
    >
      Water
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              Object {
                "fontSize": 12,
                "marginTop": 0,
              },
            ],
          ],
        ]
      }
    >
      Fresh from the tap
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              Object {
                "fontSize": 14,
              },
            ],
          ],
        ]
      }
    >
      $
      1.99
    </Text>
  </View>
</View>
`;
